lib/tunnel.js

```diff
@@ -466,24 +466,44 @@ function killRunningProcess_(that, fnCallback) {
   // use try-catch as possibility of Error while killing process.
   try {
     // Check process is running before killing
-    if (that.proc) {
-      that.proc.on('exit', function() {
-        if (that.proc.pid) {
-          setTimeout(function() {
-            try {
-              process.kill(that.proc.pid);
-            } catch (err) {}
-            return fnCallback();
-          }, 500);
+    if (that.proc && that.infoAPIPort) {
+      var url = 'http://127.0.0.1:' + that.infoAPIPort + '/api/v1.0/stop';
+      var parseURL = urlParse.parse(url);
+      var data = '';
+      var reqOptions = {
+        hostname: parseURL.hostname,
+        port: parseURL.port,
+        path: parseURL.path,
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ml-auth': 'LambdaTest',
+          'Content-Length': data.length
```

Comment:
You shouldn't add content length yourself. Let the library handle that. You can use got or axios for that

Reply:
Updated sir removed header